### PR TITLE
Add clone action to Git

### PR DESCRIPTION
**Pull request by jfryman**
This PR adds the ability for StackStorm to download a repository to a remote machine for manipulation. Used in the CI/CD workflow.

### CODE REVIEW

File: packs/git/config.yaml [hunk 1]
```diff
@@ -1,2 +1,6 @@
 ---
   url: "git@demo-git:/home/git/repos/test.git"
+
+  # For Web-Based Requests
+  user: ""
```
Comment (Kami) [inline packs/git/config.yaml]: Hm, those two settings seems to be a leftover from a github pack and are not really used here or?

**Comment by Kami** [comment]
New action LGTM, +1

File: packs/git/rules/github-pr-merge.json [hunk 2]
```diff
@@ -9,16 +9,7 @@
         }
     },
 
-    "criteria": {
```
Comment (Kami) [inline packs/git/rules/github-pr-merge.json]: I assume you have intentionally removed this section?

It looks like previously we sent all the github events to this webhook and now we only send merge events to this webook?

Comment (jfryman) [inline packs/git/rules/github-pr-merge.json]: This one I wasn't sure about. I got this change on a merge from master... it's easily revert-able. I'm not doing anything with this.

File: packs/git/config.yaml (hunk 1 above)
Comment (jfryman) [inline packs/git/config.yaml]: Sure is. :blush:. Let me get this out too. :)

File: packs/git/actions/clone.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+name: clone
+runner_type: run-remote
+description: "Clone a repository"
+enabled: true
+entry_point: ''
+parameters:
+  source:
```
Comment (manasdk) [inline packs/git/actions/clone.yaml]: I guess this only works for oublic repos unless the system user has access to these repos - right? 

I am ok with this assumption but wanted to clarify if that is your intent and how it works.

Comment (jfryman) [inline packs/git/actions/clone.yaml]: Nope, I'm using it for a private repository. It just requires that SSH Keys exist on the system. This is not how I envision the end-state to be, but it's Good Enough(tm) for now.

I would like to be able to add flags like AUTH or SSH_KEY without having to write a secondary script to handle the parameter passing and assembling the final command.

Comment (manasdk) [inline packs/git/actions/clone.yaml]: Yeah being able to provide location to a key or auth creds would really help. I agree this works for now.

**Pull request merged**